pkg/operator/tektonchains.go

```diff
@@ -0,0 +1,83 @@
+package operator
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/openshift-pipelines/release-tests/pkg/assert"
+	"github.com/openshift-pipelines/release-tests/pkg/clients"
+	"github.com/openshift-pipelines/release-tests/pkg/config"
+	"knative.dev/pkg/test/logging"
+
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/tektoncd/operator/pkg/apis/operator/v1alpha1"
+	operatorv1alpha1 "github.com/tektoncd/operator/pkg/client/clientset/versioned/typed/operator/v1alpha1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EnsureTektonChainExists creates a TektonChain with the name names.TektonChain, if it does not exist.
```

Comment:
I was confused even in other `Ensure.*Exists` function. The function does not create any resource, it just checks if the resource is present right?

Reply:
This is literally a copy&paste of `pkg/operator/tektonaddons.go`... maybe it's a good opportunity to refactor both

Reply:
I tried to track down the origin of that code and it seems that it was originally created by Nikhil in https://github.com/tektoncd/operator repo and it's still used, e.g. for pipeline
* https://github.com/tektoncd/operator/blob/release-v0.62.x/test/resources/tektonpipelines.go

versus almost identical code in release-tests repo created by Praveen in
* https://github.com/openshift-pipelines/release-tests/blob/master/pkg/operator/tektonpipelines.go

One significant difference is that original code checked if resource exists and then created it if it was necessary. In release-tests it just waits for CR to appear on cluster.

Ideally we should use directly the code from operator repo and not duplicate it in our repo.